src/search/search_algorithms/plugin_eager_greedy.cc

```diff
@@ -37,38 +39,46 @@ class EagerGreedySearchFeature : public plugins::TypedFeature<SearchAlgorithm, e
         document_note(
             "Equivalent statements using general eager search",
             "\n```\n--evaluator h2=eval2\n"
-            "--search eager_greedy([eval1, h2], preferred=h2, boost=100)\n```\n"
+            "--search eager_greedy([eval1, h2], preferred=[h2], boost=100)\n```\n"
             "is equivalent to\n"
             "```\n--evaluator h1=eval1 --heuristic h2=eval2\n"
             "--search eager(alt([single(h1), single(h1, pref_only=true), single(h2), \n"
             "                    single(h2, pref_only=true)], boost=100),\n"
-            "               preferred=h2)\n```\n"
+            "               preferred=[h2])\n```\n"
             "------------------------------------------------------------\n"
             "```\n--search eager_greedy([eval1, eval2])\n```\n"
             "is equivalent to\n"
             "```\n--search eager(alt([single(eval1), single(eval2)]))\n```\n"
             "------------------------------------------------------------\n"
             "```\n--evaluator h1=eval1\n"
-            "--search eager_greedy(h1, preferred=h1)\n```\n"
+            "--search eager_greedy([h1], preferred=[h1])\n```\n"
             "is equivalent to\n"
             "```\n--evaluator h1=eval1\n"
             "--search eager(alt([single(h1), single(h1, pref_only=true)]),\n"
-            "               preferred=h1)\n```\n"
+            "               preferred=[h1])\n```\n"
             "------------------------------------------------------------\n"
-            "```\n--search eager_greedy(eval1)\n```\n"
+            "```\n--search eager_greedy([eval1])\n```\n"
             "is equivalent to\n"
             "```\n--search eager(single(eval1))\n```\n", true);
     }
 
-    virtual shared_ptr<eager_search::EagerSearch> create_component(const plugins::Options &options, const utils::Context &context) const override {
-        plugins::verify_list_non_empty<shared_ptr<Evaluator>>(context, options, "evals");
-        plugins::Options options_copy(options);
-        options_copy.set("open", search_common::create_greedy_open_list_factory(options_copy));
-        options_copy.set("reopen_closed", false);
-        shared_ptr<Evaluator> evaluator = nullptr;
-        options_copy.set("f_eval", evaluator);
```

Comment:
We kept it if it was not trivial how to remove it.
It would be possible to get rid of more opts_copies by moving the logic into the create_component function but I think keeping it somewhat encapsulated is nice.